Source/src/components/ComponentBillboard.cpp

```diff
@@ -0,0 +1,549 @@
+#include "core/hepch.h"
+
+#include "ComponentBillboard.h"
+
+#include <imgui_color_gradient.h>
+
+#include "modules/ModuleProgram.h"
+#include "modules/ModuleResources.h"
+#include "modules/ModuleRender.h"
+#include "modules/ModuleCamera.h"
+#include "modules/ModuleSceneManager.h"
+#include "resources/ResourceTexture.h"
+#include "ComponentTransform.h"
+#include "ComponentCamera.h"
+
+Hachiko::ComponentBillboard::ComponentBillboard(GameObject* container, UID id) 
+	: Component(Component::Type::BILLBOARD, container, id) 
+{
+    gradient = new ImGradient();
+    PublishIntoScene();
+}
+
+Hachiko::ComponentBillboard::~ComponentBillboard() 
+{
+    DetachFromScene();
+    delete gradient;
+}
+
+void Hachiko::ComponentBillboard::Draw(ComponentCamera* camera, Program* program)
+{
+    glActiveTexture(GL_TEXTURE0);
+    int glTexture = 0;
```

Comment:
I missed this out, this needs to be snake_case as well, I'm sorry for noticing this now

Reply:
Done